### DISCUSSION TITLE
How can I avoid the traceback in xonsh scripts when using $RAISE_SUBPROC_ERROR = True?

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#4708] [Q&A] [answered]
In a xonsh script like

```
$RAISE_SUBPROC_ERROR = True
cp nonexisting_file.txt other_name.txt
print("after the failure")
```

I do not get to the last line (`print("after the failure")`). Instead I get message`subprocess.CalledProcessError: Command '['cp', 'nonexisting_file.txt',  'other_name.txt']' returned non-zero exit status 1`.

That's great. Exactly what I want. 😄 However before the message I get the whole stack trace. How can I avoid this? I tried `$XONSH_SHOW_TRACEBACK = False` in the script, but that doesn't seem to help.

**Comment by xmnlab** [comment]
hi everyone! I have the same problem, I would love to know if there is a way to workaround this problem

**Comment by anki-code** [comment]
Here should be the fix (short traceback in case `$XONSH_SHOW_TRACEBACK = False`):

https://github.com/xonsh/xonsh/blob/eae3fcc255d9c540b3fae34a35e659f5fd7c1e7d/xonsh/procs/pipelines.py#L597-L599

PR is welcome!

**Reply by xmnlab** [reply]
A PR is coming :) https://github.com/xonsh/xonsh/pull/5066
a review would be very appreciated. it is my first time contributing here .. so I am not super familiar with the code base yet

**Reply by anki-code** [reply]
Thanks and welcome! I already commented the PR )

**Comment by xmnlab** [comment, marked as answer]
this issue is already fixed on the master branch.
@anki-code do you have any idea about when a next release will be published?

**Reply by anki-code** [reply]
Usually release going quarterly but it depends on maintainers load.
You can install xonsh and use from master branch easily with pip or xpip:
```xsh
xpip install -U git+https://github.com/xonsh/xonsh
# restart xonsh
```

**Reply by xmnlab** [reply]
thanks for the information!
